### DISCUSSION TITLE
How to capture Datagrid ColumnWidth_Changed event?

### DISCUSSION THREAD

**Discussion by yaoyulian** [AvaloniaUI/Avalonia#9511] [Q&A] [answered]
I hope save Datagrid ColumnWidth to disk after user changed ColumnWidth.  Then I can restore ColumnWidth at App reopen.
In WPF，like this:
`DependencyPropertyDescriptor descriptor = DependencyPropertyDescriptor.FromProperty(DataGridColumn.WidthProperty, typeof(DataGridColumn));
descriptor.AddValueChanged(datagrid.Columns[1], ColumnWidth_Changed);
private void ColumnWidth_Changed(object sender, EventArgs e)
{
SaveWidth();
}`
How to capture Datagrid ColumnWidth_Changed event in Avalonia?
I have tryed these, but not validly :
`datagrid.PropertyChanged += Datagrid_PropertyChanged;
datagrid.PointerCaptureLost += Datagrid_PointerCaptureLost;
datagrid.Columns[1].PropertyChanged += DatagridColumns_PropertyChanged;
datagrid.PointerPressed`

**Comment by timunie** [comment, marked as answer]
@yaoyulian can't you just bind to properities in your ViewModel?

**Reply by yaoyulian** [reply]
bind have another problem [https://github.com/AvaloniaUI/Avalonia/issues/4326](https://github.com/AvaloniaUI/Avalonia/issues/4326)
`<DataGridTextColumn`  Width="{Binding $parent[Window].FieldColumnWidth}" />
Avalonia error XAMLIL: Unable to find suitable setter or adder for property Width of type Avalonia.Controls.DataGrid:Avalonia.Controls.DataGridColumn for argument Avalonia.Markup:Avalonia.Data.Binding, available setter parameter lists are:`
